packages/now-cli/src/util/link/setup-and-link.ts

```diff
@@ -0,0 +1,225 @@
+import { join, basename } from 'path';
+import chalk from 'chalk';
+import { remove } from 'fs-extra';
+import { NowContext, ProjectLinkResult } from '../../types';
+import { NowConfig } from '../dev/types';
+import { Output } from '../output';
+import {
+  getLinkedProject,
+  linkFolderToProject,
+  getVercelDirectory,
+} from '../projects/link';
+import createProject from '../projects/create-project';
+import updateProject from '../projects/update-project';
+import Client from '../client';
+import handleError from '../handle-error';
+import confirm from '../input/confirm';
+import toHumanPath from '../humanize-path';
+import { isDirectory } from '../config/global-path';
+import selectOrg from '../input/select-org';
+import inputProject from '../input/input-project';
+import { validateRootDirectory } from '../validate-paths';
+import { inputRootDirectory } from '../input/input-root-directory';
+import editProjectSettings from '../input/edit-project-settings';
+import stamp from '../output/stamp';
+import { EmojiLabel } from '../emoji';
+//@ts-expect-error
+import createDeploy from '../deploy/create-deploy';
+//@ts-expect-error
+import Now from '../index';
+
+export default async function setupAndLink(
+  ctx: NowContext,
+  output: Output,
+  path: string,
+  force: boolean,
+  successEmoji: EmojiLabel
+): Promise<ProjectLinkResult> {
+  const {
+    authConfig: { token },
+    config,
+  } = ctx;
+  const { apiUrl } = ctx;
+  const debug = output.isDebugEnabled();
+  const client = new Client({
+    apiUrl,
+    token,
+    currentTeam: config.currentTeam,
+    debug,
+  });
+
+  if (!isDirectory(path)) {
+    output.error(`Expected directory but found file: ${path}`);
+    return { status: 'error', exitCode: 1 };
+  }
+  const link = await getLinkedProject(output, client, path);
+  const autoConfirm = false;
+  const isFile = false;
+  const isTTY = process.stdout.isTTY;
+  const quiet = !isTTY;
+  let rootDirectory = null;
+  let newProjectName = null;
+  let org;
+
+  if (!force && link.status === 'linked') {
+    return link;
+  }
+
+  if (force) {
+    const vercelDir = getVercelDirectory(path);
+    remove(vercelDir);
+  }
+
+  const shouldStartSetup =
+    autoConfirm ||
```

Comment:
`autoConfirm` is always `false`. Is that expected?

Reply:
I think this is used with `--confirm` so maybe we should add `vc link --confirm` and `vc dev --confirm`? What do you think?

Reply:
That'd would make sense to me 👍

Reply:
I added `vc link --confirm` and `vc dev --confirm`.